### ISSUE TITLE
Docs: CLI --config example uses legacy example filename

### ISSUE THREAD

**Issue by MikeMcC399** [eslint/eslint#18921]
### Docs page(s)

[Basic Configuration> -c, --config](https://eslint.org/docs/latest/use/command-line-interface#-c---config)

### What documentation issue do you want to solve?

[Basic Configuration> -c, --config](https://eslint.org/docs/latest/use/command-line-interface#-c---config) refers to using a legacy `my-eslint.json` config file. Attempting to use this with ESLint 9.x invokes an error message if `ESLINT_USE_FLAT_CONFIG` is undefined:

```text
TypeError [ERR_IMPORT_ASSERTION_TYPE_MISSING]
```

### What do you think is the correct solution?

It should be using ESLint 9 [Configuration File](https://eslint.org/docs/latest/use/configure/configuration-files#configuration-file) names, not a filename from [Configuration Files (Deprecated)](https://eslint.org/docs/latest/use/configure/configuration-files-deprecated).

The example could however also be extended to show how to use both deprecated and flat configuration files.

### Participation

- [ ] I am willing to submit a pull request for this change (unchecked box means NO).

**Comment by nzakas** [comment]
Thanks. Do you want to submit a PR to make this change?

**Comment by MikeMcC399** [comment]
@nzakas 

Thank you for your acceptance of the documentation issue.

> Do you want to submit a PR to make this change?

I purposely didn't select `I am willing to submit a pull request for this change.` option. I'm happy for somebody with the right level of skills to handle the resolution. That wouldn't be me though. 🙂

**Comment by nzakas** [comment]
There's an "Edit Page" button at the bottom of the page you saw this one. :smile: Everything can be done right in the browser.

**Comment by MikeMcC399** [comment]
@nzakas 

I'm afraid I was not clear enough.  I'm saying that I don't feel skilled enough in ESLint to want to be providing documentation. I am well skilled in git / GitHub etc. That is not the issue.

**Comment by amareshsm** [comment]
> I'm afraid I was not clear enough. I'm saying that I don't feel skilled enough in ESLint to want to be providing documentation. I am well skilled in git / GitHub etc. That is not the issue.

You can still give it a try with what you know! The team will guide you during the PR process. If it doesn’t work out, no worries at all.

**Comment by Jay-Karia** [comment]
I am working on #18784 which deals with all the commands, so should I change this command in that PR?

To: `npx eslint -c ~/eslint.config.json file.js`

**Comment by MikeMcC399** [comment]
@Jay-Karia 

I would expect the following replacement command you are proposing to fail in ESLint 9 with an error 

`TypeError [ERR_IMPORT_ASSERTION_TYPE_MISSING]`

```shell
npx eslint -c ~/eslint.config.json file.js
```

If you replace the command instead with

```shell
ESLINT_USE_FLAT_CONFIG=false npx eslint -c ~/eslint.config.json file.js
```

then I would expect you would receive instead a warning

```text
(node:5728) ESLintRCWarning: You are using an eslintrc configuration file, which is deprecated and support will be removed in v10.0.0. Please migrate to an eslint.config.js file. See https://eslint.org/docs/latest/use/configure/migration-guide for details.
(Use `node --trace-warnings ...` to show where the warning was created)
```

Considering that this section needs content rework, there doesn't seem to be any advantage in replacing one non-working command with another non-working command, so I suggest not to change it at this time.

- Good luck with your PR #18784!

**Comment by Jay-Karia** [comment]
As I was looking through the docs `eslint.config.json` is not supported: https://eslint.org/docs/latest/use/configure/configuration-files#configuration-file

We can do `npx eslint -c ~/eslint.config.js file.js`
And change the content below it

**Comment by MikeMcC399** [comment]
@Jay-Karia 

> As I was looking through the docs `eslint.config.json` is not supported: https://eslint.org/docs/latest/use/configure/configuration-files#configuration-file

I don't believe it is correct to say that `eslint.config.json` is not supported. 
https://eslint.org/docs/latest/use/configure/configuration-files-deprecated indicates that it is deprecated and supported in ESLint 9.

**Comment by MikeMcC399** [comment]
@Jay-Karia 

It appears to me that the example of using a configuration file in the user's home directory `~` only makes sense in legacy configurations where configurations can be merged. My understanding of flat configuration is that there is no merging done anymore. So I don't know if the replacement example `npx eslint -c ~/eslint.config.js file.js` has any practical use.

**Comment by amareshsm** [comment]
> I am working on [#18784](https://github.com/eslint/eslint/pull/18784) which deals with all the commands, so should I change this command in that PR?
> 
> To: `npx eslint -c ~/eslint.config.json file.js`

@Jay-Karia You can submit this change as a separate PR, which will make it easier for reviewers to provide feedback.

**Comment by Jay-Karia** [comment]
Ok, will open after #18784 gets merged, to avoid conflicts.

**Comment by chetanpilane** [comment]
can i work on this issue?

**Comment by mdjermanovic** [comment]
@chetanpilane of course, feel free to submit a PR.

**Comment by Danesh-md** [comment]
hi this is my first open source project. I would like to fix this issue. Idk how to start, can anyone guide me with this?

**Comment by MikeMcC399** [comment]
Now I'm confused about who will do the work.

- The issue is assigned to @amareshsm 
- @Jay-Karia has said they would submit a PR
- @chetanpilane has been encouraged to also submit a PR

**Comment by Jay-Karia** [comment]
I am less active on GitHub nowadays due to some reasons, so won't be able to complete this task.

**Comment by MikeMcC399** [comment]
@Jay-Karia 

> I am less active on GitHub nowadays due to some reasons, so won't be able to complete this task.

Many thanks for your clarification and all the best to you!

**Comment by Danesh-md** [comment]
Can u assign me @MikeMcC399 ??
And can you tell what I need to know to fix this issue?

**Comment by MikeMcC399** [comment]
@Danesh-md 

> Can u assign me [@MikeMcC399](https://github.com/MikeMcC399) ?? And can you tell what I need to know to fix this issue?

I'm afraid that there is a misunderstanding. I am only the submitter of the issue. I cannot assign the issue to anybody. That would need to be done by the maintainers of this repo and I am not a maintainer.

**Comment by Danesh-md** [comment]
Oh okay bro got it

**Comment by Danesh-md** [comment]
@MikeMcC399 sir I'm eager to start contributing to opensource but not sure where to begin. If you have any tips or advice, I'd really appreciate it sir! Could I also connect with you on LinkedIn or Instagram? Thank you for your time!

**Comment by MikeMcC399** [comment]
@Danesh-md 

> sir I'm eager to start contributing to opensource to make my resume strong but not sure where to begin. If you have any tips or advice, I'd really appreciate it sir! Could I also connect with you on LinkedIn or Instagram? Thank you for your time!

I suggest you Google "how to contribute to open source" for instance https://opensource.guide/how-to-contribute/ , however I am not a mentor for this so you'll have to find your own way. It's also off-topic to raise such a request here.

**Comment by Danesh-md** [comment]
Thankyou sir. Sorry for raising the question here. I will delete my comment sir.

**Comment by mdjermanovic** [comment]
Here are the guidelines:

https://eslint.org/docs/latest/contribute/pull-requests

**Comment by VISHESH0932** [comment]
Hey @amareshsm ,is there any work need to be done?

**Comment by amareshsm** [comment]
@VISHESH0932  pls check the description and the other discussion in this issue. f you're interested in working on it, let us know, and we can assign it to you.

**Comment by JP-sDEV** [comment]
If this is still open and @VISHESH0932 does not respond, I will gladly take this issue. Cheers! :)

**Comment by VISHESH0932** [comment]
I am unable to understand the issue as i am new to it.
@JP-sDEV  go ahead
btw thanks @amareshsm.
